jdaviz/tests/test_subsets.py

```diff
@@ -566,20 +566,26 @@ def test_edit_composite_spectral_with_xor(specviz_helper, spectrum1d):
     specviz_helper.load_data(spectrum1d)
     viewer = specviz_helper.app.get_viewer(specviz_helper._default_spectrum_viewer_reference_name)
 
-    viewer.apply_roi(XRangeROI(6400, 6600))
+    viewer.apply_roi(XRangeROI(6200, 6800))
     specviz_helper.app.session.edit_subset_mode.mode = OrMode
-    viewer.apply_roi(XRangeROI(7200, 7400))
-
-    viewer.apply_roi(XRangeROI(7600, 7800))
+    viewer.apply_roi(XRangeROI(7200, 7600))
 
     specviz_helper.app.session.edit_subset_mode.mode = XorMode
-    viewer.apply_roi(XRangeROI(6700, 7700))
+    viewer.apply_roi(XRangeROI(6100, 7600))
```

Comment:
The bug gets triggered when XOR is not exactly the same limits as the other regions (and that is usually the case in practice).